rpcd/playbooks/rpc-pre-upgrades.yml

```diff
@@ -19,18 +19,63 @@
   roles:
     - { role: "rpc_pre_upgrade", tags: [ "rpc-pre-upgrades" ] }
 
-- name: restart beaver
+- name: remove beaver
   hosts: all
   tags:
     - rpc-pre-upgrades
+    - remove-beaver
   tasks:
     - name: Check if beaver is installed
       stat:
         path: /etc/init.d/beaver
       register: beaver_init_script
+      changed_when: false
 
-    - name: Restart beaver if installed
+    - name: Make sure beaver is stopped
       service:
         name: beaver
-        state: restarted
+        enabled: no
+        state: stopped
       when: beaver_init_script.stat.exists
+
+    - name: Remove beaver etc files
+      file:
+        path: /etc/beaver
+        state: absent
+
+    - name: Remove beaver init files
+      file:
+        path: /etc/init.d/beaver
+        state: absent
+
+    - name: Remove beaver logrotate files
+      file:
+        path: /etc/logrotate.d/beaver
+        state: absent
+
+    - name: Remove beaver log files
+      file:
+        path: /var/log/beaver
+        state: absent
+
+    - name: Check if beaver is installed
+      shell: pip show -q Beaver
```

Comment:
If you're not using shell specific features, don't use the `shell` module, use `command`. That said, while you could just add ` || true` to the end of the command with the `shell` module, there's a bigger issue. There's lot of types of errors that could occur aside from just beaver not being installed (pip not installed, pip failing, syntax issues in the args, path issues, environment/locale/encoding issues, etc).

I think this can be done way better than just hamfisting "never fails" and "never changes". For one thing, there's an `ignore_errors` key that can be set, which allows the check task to fail, but a slightly better way is to break this into smaller chunks. Example:

command: pip list
register: pip_packages

pip:
  name: beaver
  state: absent
when: pip_packages.stdout.find('Beaver') != -1

But! Since your goal is to uninstall the package anyway using the `pip` module in the next task, none of this even necessary!

Ansible modules are designed to be idempotent. And we can even confirm that the `pip` module works this with when uninstalling: https://github.com/ansible/ansible-modules-core/blob/stable-2.1/packaging/language/pip.py#L413

Meaning you can get rid of the shell command and checks entirely. All you need is this:

pip:
  name: beaver
  state: absent

Just as $deity intended :)